File: Review-06-CPP-Shapes/Example-2/source/square.cpp
// Thomas Kennedy
// CS 330 Fall 2014

#include <iomanip>

#include "square.h"

/**
 *
 */
Square::Square()
{
    this->_name = "Square";
    this->_side = 1;
}

/**
 *
 */
Square::Square( double s )
{
    this->_name = "Square";
    this->_side = s;
}

/**
 *
 */
Square::Square( const Square &src )
{
    this->_name = src._name;
    this->_side = src._side;
}

/**
 *
 */
Square::~Square()
{
}

/**
 *
 */
void Square::display( std::ostream &outs ) const
{
    Shape::display(outs);

    outs << std::left << std::setw( WIDTH_LABEL ) << "Side"       << ": " << std::right << std::setw( WIDTH_VALUE ) << side()       << "\n"
         << std::left << std::setw( WIDTH_LABEL ) << "Perimeter"  << ": " << std::right << std::setw( WIDTH_VALUE ) << perimeter()  << "\n"
         << std::left << std::setw( WIDTH_LABEL ) << "Area"       << ": " << std::right << std::setw( WIDTH_VALUE ) << area()       << "\n";
}

File: Review-06-Java-Shapes/Example-9/src/main/java/edu/odu/cs/cs330/examples/shapes/Square.java
// Thomas Kennedy
// CS 330 Fall 2020

package edu.odu.cs.cs330.examples.shapes;

/**
 * A Rectangle with 4 Equal Sides.
 *
 * @author Thomas J Kennedy
 */
public class Square implements Shape {
    /**
     * Length of One Side.
     */
    private double _side;

    /**
     * Construct a Square with side set to 1.
     */
    public Square()
    {
        this(1);
    }

    /**
     * Construct a Square.
     *
     * @param s the desired side length
     */
    public Square(double s)
    {
        _side = s;
    }

    /**
     * Return the side length.
     *
     * @return length of a single side
     */
    public double side()
    {
        return _side;
    }

    /**
     * Modify the side length.
     *
     * @param s the replacement length
     */
    public void side(double s)
    {
        _side = s;
    }

    @Override
    public String name()
    {
        return "Square";
    }

    @Override
    public double area()
    {
        return Math.pow(_side, 2.0);
    }

    @Override
    public double perimeter()
    {
        return 4 * _side;
    }

    @Override
    public Object clone() throws CloneNotSupportedException
    {
        return new Square(this.side());
    }

    @Override
    public int numDims()
    {
        return 1;
    }

    @Override
    public void createFromDims(double[] dims)
    {
        this._side = dims[0];
    }

    @Override
    public String toString()
    {
        return String.join(
            "",
            String.format(Shape.FMT_STR, "Name", this.name()),
            String.format(Shape.FMT_DBL, "Side", this._side),
            String.format(Shape.FMT_DBL, "Perimeter", this.perimeter()),
            String.format(Shape.FMT_DBL, "Area", this.area())
        );
    }
}





